File: backend/src/main/java/com/springbootjwtpostgres/backend/inventory/Inventory.java
package com.springbootjwtpostgres.backend.inventory;

import com.springbootjwtpostgres.backend.basemodels.BaseEntity;
import com.springbootjwtpostgres.backend.product.Product;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.OneToOne;

@Entity
@EqualsAndHashCode(callSuper = true)
@Data
@NoArgsConstructor
public class Inventory extends BaseEntity {
    private Long currentQuantity;

    @OneToOne
    private Product product;
}

File: frontend/src/app/models/inventory/Inventory.ts
import { Product } from '../product/Product';
import { BaseEntity } from '../base/BaseEntity';

export class Inventory extends BaseEntity {
  currentQuantity: number;
  product: Product;

  constructor(currentQuantity: number, product: Product) {
    super();
    this.currentQuantity = currentQuantity;
    this.product = product;
  }
}
